include/eve/detail/function/simd/riscv/friends.hpp

```diff
@@ -0,0 +1,282 @@
+{
+  return __riscv_vmseq(lhs, rhs, N::value);
+}
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE auto
+self_eq(wide<T, N> lhs, std::convertible_to<T> auto rhs) noexcept -> logical<wide<T, N>>
+requires rvv_abi<abi_t<T, N>> && (!std::is_floating_point_v<T>)
+{
+  return __riscv_vmseq(lhs, static_cast<T>(rhs), N::value);
+}
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE auto
+self_eq(wide<T, N> lhs, wide<T, N> rhs) noexcept -> logical<wide<T, N>>
+requires rvv_abi<abi_t<T, N>> && (std::is_floating_point_v<T>)
+{
+  return __riscv_vmfeq(lhs, rhs, N::value);
+}
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE auto
+self_eq(wide<T, N> lhs, std::convertible_to<T> auto rhs) noexcept -> logical<wide<T, N>>
+requires rvv_abi<abi_t<T, N>> && (std::is_floating_point_v<T>)
+{
+  return __riscv_vmfeq(lhs, static_cast<T>(rhs), N::value);
+}
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE auto
+self_neq(wide<T, N> lhs, wide<T, N> rhs) noexcept -> logical<wide<T, N>>
+requires rvv_abi<abi_t<T, N>> && (!std::is_floating_point_v<T>)
+{
+  return __riscv_vmsne(lhs, rhs, N::value);
+}
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE auto
+self_neq(wide<T, N> lhs, std::convertible_to<T> auto rhs) noexcept -> logical<wide<T, N>>
+requires rvv_abi<abi_t<T, N>> && (!std::is_floating_point_v<T>)
+{
+  return __riscv_vmsne(lhs, static_cast<T>(rhs), N::value);
+}
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE auto
+self_neq(wide<T, N> lhs, wide<T, N> rhs) noexcept -> logical<wide<T, N>>
+requires rvv_abi<abi_t<T, N>> && (std::is_floating_point_v<T>)
+{
+  return __riscv_vmfne(lhs, rhs, N::value);
+}
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE auto
+self_neq(wide<T, N> lhs, std::convertible_to<T> auto rhs) noexcept -> logical<wide<T, N>>
+requires rvv_abi<abi_t<T, N>> && (std::is_floating_point_v<T>)
+{
+  return __riscv_vmfne(lhs, static_cast<T>(rhs), N::value);
+}
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE auto
+self_eq(logical<wide<T, N>> lhs, logical<wide<T, N>> rhs) noexcept -> logical<wide<T, N>>
+requires rvv_abi<abi_t<T, N>>
+{
+  return __riscv_vmxnor(lhs, rhs, N::value);
+}
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE auto
+self_neq(logical<wide<T, N>> lhs, logical<wide<T, N>> rhs) noexcept -> logical<wide<T, N>>
+requires rvv_abi<abi_t<T, N>>
+{
+  return __riscv_vmxor(lhs, rhs, N::value);
+}
+
+template<typename T, typename U, typename N>
+EVE_FORCEINLINE auto
+self_logand(rvv_ const&, logical<wide<T, N>> v, logical<wide<U, N>> w) noexcept
+    -> logical<wide<T, N>>
+requires(rvv_abi<abi_t<T, N>> || rvv_abi<abi_t<U, N>>)
+{
+  if constexpr( !is_aggregated_v<abi_t<T, N>> && !is_aggregated_v<abi_t<U, N>> )
+  {
+    auto                casted_w = bit_cast(w, as<logical<wide<T, N>>> {});
+    logical<wide<T, N>> to_ret   = __riscv_vmand(v, casted_w, N::value);
+    return to_ret;
+  }
```

Comment:
Here I've added this, as in logical's that I've created for RISC-V are different. Suppose you want to fully fill `char` `wide` with vlen=128 and lmul = 8. It will be `wide<char, 128>`. But for `int`, the maximum `wide`, that could be placed on physical registers is `wide<int, 32>` (128 / 32 * 8). If we have logical, that could be effectively work with this type of wide, even if it can actually "fit" into native vector register mask type, I split them. Otherwise, operations for aggregated `wide` with predicate will not be cheap (I will have to construct real mask types on place).